Java/Java.sublime-syntax

```diff
@@ -230,89 +180,98 @@ contexts:
+    - include: class
+    # Get modifiers defined on a different line than the class
     - include: storage-modifiers
-    - match: (\bclass|(?:@)?\binterface|\benum)(?:\s+|\b)
-      captures:
-        1: storage.type.java
-      set:
-        - match: '\w+'
-          scope: meta.class.identifier.java entity.name.class.java
-          set: class-declaration
-        - match: ''
-          set: class-declaration
-    - match: '<'
-      scope: punctuation.definition.generic.begin.java
+    - include: stray-braces
+  class:
+    - match: '(?=\w?[\w\s]*(?:class|(?:@)?interface|enum)\s+\w+)'
       push:
-        - meta_scope: meta.generic.java
-        - match: '>'
-          scope: punctuation.definition.generic.end.java
-          pop: true
-        - include: generic-instantiation
-    - match: (?=\bextends\b)
-      set:
+        - meta_scope: meta.class.java
+        # Needed because the modifers are part of the meta scope
+        - include: storage-modifiers
+        - include: generic-declaration
+        - match: (\bclass|(?:@)?\binterface|\benum)\s+(\w+)
+          scope: meta.class.identifier.java
+          captures:
+            1: storage.type.java
+            2: entity.name.class.java
         - match: \bextends\b
           scope: keyword.declaration.extends.java
-          set:
+          push:
             - meta_scope: meta.class.extends.java
             - match: '(?=\s*{|\s*\bimplements\b)'
-              set: class-declaration
+              pop: true
             - include: object-types-inherited
-    - match: (?=\bimplements\b)
-      set:
-        - match: \bimplements\b
+        - match: \b(implements)\b
           scope: keyword.declaration.implements.java
-          set:
+          push:
             - meta_scope: meta.class.implements.java
-            - match: '(?=\s*\bextends\b|\s*\{)'
-              set: class-declaration
+            - match: '(?=\s*{|\s*\bextends\b)'
+              pop: true
+            - match: \,
+              scope: punctuation.separator.implements.java
             - include: object-types-inherited
-    - match: '(?=\{)'
-      set:
-        - meta_scope: meta.class.body.java
-        - match: '\}'
-          scope: punctuation.section.class.end.java
-          pop: true
-        - match: '\{'
+        - match: "{"
           scope: punctuation.section.class.begin.java
           push:
-            - match: "(?=})"
+            - meta_scope: meta.class.body.java
+            - match: "}"
+              scope: punctuation.section.class.end.java
               pop: true
             - include: class-body
+        - include: any_POP
 
   class-body:
+    - include: annotations
     - include: class
+    - include: static-assignment
     - include: enums
+    - include: generic-declaration # for methods
     - include: methods
-    - include: annotations
+    # mark constants as constant
+    - match: '(\.)?\b(?!UUID\b|URI\b)([A-Z][A-Z0-9_]+)(?!<|\.class|\s*\w+\s*=)\b'
```

Comment:
Yeah they are present in the standard java API. I should extract that out into a variable. I should also add tests. 
They break the standard class naming convention. Without this they get picked up as constants.